classes/db/Db.php

```diff
@@ -567,10 +567,10 @@ public function executeS($sql, $array = true, $use_cache = true)
 		if ($use_cache && $this->is_cache_enabled && $array)
 		{
 			$this->last_query_hash = Tools::encryptIV($sql);
-			if (($result = Cache::getInstance()->get($this->last_query_hash)) !== false)
+			if (Cache::getInstance()->exists($this->last_query_hash) !== false)
```

Comment:
@mdomenjoud : actually it would also work for APC / Memcached, but would be more expensive than the current implementation (for example in memcached implementation, the exists function is just an alias to the get function because it doesn't exist). I need to do some extra benchmark, but I fear cacheFS is really inefficient (it unserialize/serialize potentially *really* big array), and the extra lock added will increase this inefficiency, so I would not recommend using it *at all*